File: atlas/examples/kotlin/domain/src/main/kotlin/dev/atlas/example/domain/catalog/CatalogItemDraft.kt
package dev.atlas.example.domain.catalog

/**
 * Carries validated catalog seed data into the domain factory.
 *
 * @property id External identifier candidate.
 * @property title External title candidate.
 * @property kind Concrete item specialization.
 * @property contributor Author or facilitator display name.
 */
data class CatalogItemDraft(
    val id: String,
    val title: String,
    val kind: CatalogItemKind,
    val contributor: String
)

File: atlas/examples/typescript/packages/domain/src/catalog/CatalogItemDraft.ts
import type { CatalogItemKind } from './CatalogItemKind.js';

/**
 * Carries validated catalog seed data into the domain factory.
 */
export interface CatalogItemDraft {
  readonly id: string;
  readonly title: string;
  readonly kind: CatalogItemKind;
  readonly contributor: string;
}
